packages/typescript-estree/tests/lib/convert.test.ts

```diff
@@ -361,4 +361,13 @@ describe('convert', () => {
       expect(Object.keys(esCallExpression)).toContain('typeParameters');
     });
   });
+  describe('using should be forbidden in for-related initializer ', () => {
+    it('using should be forbidden in for in statement ', () => {
+      const ast = convertCode('for(using foo in {});');
+
+      const instance = new Converter(ast);
+
+      expect(() => instance.convertProgram()).toThrow();
+    });
+  });
```

Comment:
We don't put AST tests in this location - this is more for testing our infra.
Instead we place our AST tests in the `ast-spec` package alongside the AST declarations.

We haven't written proper docs for this yet (sorry!) but the steps for doing this is as follows:

1) within [this folder](https://github.com/typescript-eslint/typescript-eslint/tree/main/packages/ast-spec/src/statement/ForInStatement) create a folder called `fixtures`.
2) within `fixtures` create a `_error_` folder
3) within `_error_` create a folder with a name for your case - for example maybe `using-initializer`?
4) within that folder, create a file called `fixture.ts`
5) within that file add your test code (eg `for(using foo in {});`)
6) `cd packages/ast-spec && yarn test` to run the fixture tests and generate the necessary snapshots.